### PR TITLE
Add corner case to norm

### PR DESCRIPTION
**Pull request by msluszniak**

### CODE REVIEW

**Comment by jonatanklosko** [comment]
Let's add a doctest for this case as well :)

**Comment by josevalim** [comment]
@msluszniak note the tests are failing. :)

**Review by polvalente — APPROVED** [review]

**Comment by msluszniak** [comment]
> @msluszniak note the tests are failing. :)

Yeah, I forget that this function is not defn and `if` evaluates wrongly.

File: nx/lib/nx/lin_alg.ex [hunk 1]
```diff
@@ -358,7 +364,8 @@ defmodule Nx.LinAlg do
     # The idea is that by dividing the tensor by it, large values of
     # tensor entries and large values of p are reduced, which in turn
     # avoids numerical overflow.
-    numerical_stability_coefficient = Nx.reduce_max(abs_t)
+    # The second max prevents from division by zero.
+    numerical_stability_coefficient = Nx.reduce_max(abs_t) |> Nx.max(1.0e-16)
```
Comment (josevalim) [inline nx/lib/nx/lin_alg.ex:368]: Maybe a select is better?

```suggestion
    numerical_stability_coefficient = Nx.reduce_max(abs_t)
    numerical_stability_coefficient = Nx.select(Nx.greater_than(numerical_stability_coefficient, 0), numerical_stability_coefficient, 1)
```

Comment (msluszniak) [reply nx/lib/nx/lin_alg.ex]: Yes, I changed the code to a version with separate defnp, but this version is actually the best.

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**